core/lb.go

```diff
@@ -80,8 +89,18 @@ func (lb *LoadBalancingTranscoder) createSession(md *SegTranscodingMetadata) (*t
 	// Acquire transcode session. Map to job id + assigned transcoder
 	key := job + "_" + transcoder
 	costEstimate := calculateCost(md.Profiles)
+	var lpmsSession TranscoderSession
+	if md.DetectorEnabled {
+		var err error
+		lpmsSession, err = lb.newDetectorT(DetectorProfile, transcoder)
```

Comment:
Hardcode detection device to "0"
Thanks @yondonfu for catching this!

Reply:
Made this change in https://github.com/livepeer/go-livepeer/pull/1979/commits/c17d16e113775f0957bcaefb41a65b66dd756ea6 in `core/transcoder.go` as we'd still want to schedule the transcoding on the GPU that load balancer selected - but only the detection filter hardcoded to device "0".